class TooManyTries(Exception):
    pass
class CanceledByUser(Exception):
    pass